src/vs/vscode.d.ts

```diff
@@ -5813,6 +5813,28 @@ declare module 'vscode' {
 		tooltip?: string;
 	}
 
+	/**
+	 * Provides a terminal profile for the contributed terminal profile when launched via the UI or
+	 * command.
+	 */
+	export interface TerminalProfileProvider {
+		/**
+		 * Provide the terminal profile.
+		 * @param token A cancellation token that indicates the result is no longer needed.
+		 */
+		provideTerminalProfile(token: CancellationToken): ProviderResult<TerminalProfile>;
```

Comment:
What happens if a provider returns `null` or `undefined`? Do we fallback to the default options? 

Probably worth adding a note about this to the documentation

Reply:
I think right now it will just fail. Should we really be calling out what happens when invalid values are provided such that they wouldn't even compile?

Reply:
Added an error notification for this, I don't think we should describe the failure behavior.

Reply:
`ProviderResult` explicitly allows returning `undefined` or `null`. If you don't want to support this, use `: TerminalProfile | Thenable<TerminalProfile>` instead

Reply:
Good point, will fix that up